build/Dockerfile

```diff
@@ -12,6 +12,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+FROM registry.access.redhat.com/ubi9/python-311 as wheel
+
+ARG WHEEL_VERSION=""
+USER root
+RUN --mount=type=cache,target=/root/.cache/pip \
+    python -m pip install --upgrade pip && \
+    python -m pip install build
+COPY tuning tuning
+RUN --mount=source=.git,target=.git,type=bind \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    # build wheel if wheel version is empty else download the wheel from PyPi
+    if [[ -z "${WHEEL_VERSION}" ]]; \
+    then python -m build --wheel --outdir /tmp; \
+    else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps; \
+    fi && \
```

Comment:
@alex-jw-brooks @anhuong @jbusche This change optionally allows install versioned whl. For example, you can try:
```
docker build --build-arg WHEEL_VERSION=0.0.1 --progress=plain -t fms-hf-tuning:0.0.1 . -f build/Dockerfile
```